kratos/sources/constitutive_law.cpp

```diff
@@ -447,6 +447,22 @@ array_1d<double, 6 > & ConstitutiveLaw::CalculateValue(Parameters& rParameterVal
     return rValue;
 }
 
+void ConstitutiveLaw::CalculateDerivative(Parameters& rParameterValues, const Variable<double>& rFunctionVariable, const Variable<double>& rDerivativeVariable, double& rOutput)

```

Comment:
```suggestion
void ConstitutiveLaw::CalculateDerivative(
    Parameters& rParameterValues,
    const Variable<double>& rFunctionVariable,
    const Variable<double>& rDerivativeVariable,
    double& rOutput)

```

Reply:
(same in the other methods)

Reply:
I also wonder if we should throw an error if you end up here since we will be silently returning the `rOutput` as it is.

Reply:
or return zero? Because theoratically if the developer has done a good job, then he/she should have captured all the dependent variables, so if someone asks for a derivative which it does not depend on, we can return zero version of each data type?

Reply:
My point is that if someone asks for a non implemented derivative and we return zero, we are silently hiding an error in the formulation. On the contrary, if we throw a meaningful error saying that "Derivative of rFunctionVariable.Name() w.r.t. rDerivativeVariable.Name() is not implemented" he/she well be aware of that.

Reply:
done :)